sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -1022,13 +1022,25 @@ trait CheckAnalysis extends LookupCatalog with QueryErrorsBase with PlanToString
       }
     }
 
+    def checkExistsDefaultDeterministic(colsToAdd: Seq[QualifiedColType]): Unit = {
+      colsToAdd.foreach { col =>
+        col.default.foreach { d =>
+          if (!d.deterministic) {
+            throw QueryCompilationErrors.defaultValueNotConstantError(
```

Comment:
but the check is for deterministic, not constant?

Reply:
this is to pass the test:  https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala#L3564

sorry do you have thought what check I can do instead?  If i change it to '!d.foldable', it wont work because optimizer hasnt run yet and we have expressions like 1 + 2.

if we dont have this here, we get the generic error 'INVALID_NON_DETERMINISTIC_EXPRESSIONS' from CheckAnalysis (https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala#L887)

Reply:
btw, i moved this to ColumnDefinition, but as is, didnt find a good way to change the check

Reply:
but `1 + 2` is a foldable expression, and previously we also used the `.foldable` check: https://github.com/apache/spark/pull/50701/files#diff-e28f12190a8d2f42328550508b1021fb22222ac924065aa2a769da68a51bb3beL145

Reply:
Whoops i think i tried d.foldable.  It should be d.child.foldable.  I made the change, let me try it and see if its better.

Reply:
actually this one fails  https://github.com/apache/spark/pull/50701/files#diff-1ca52084cb1c8f816d6002d507797e46ca495db46e0328220a168b6a59b4af60R528  (current_catalog() as default value)

I think the previous version ran 'FinishAnalysis', which replaced the 'current_xxx' before the foldable check

Reply:
OK , making another try, now I remove both checks (this one, and escape the existing one :  'INVALID_NON_DETERMINISTIC_EXPRESSIONS' from CheckAnalysis (https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala#L887), and rely on the later runtime toV2Col check.